The application '--version' does not exist

Today `dotnet --version` stopped working for me.

I use a [Github Actions workflow](https://github.com/iainelder/dotfiles/blob/160fce5022b72291c3aa0eb5c6c8f6b79818aa74/.github/workflows/dotnetsdk.yml) to test my script to install dotnet-sdk-7.0 on Ubuntu 22.

[My script](https://github.com/iainelder/dotfiles/blob/160fce5022b72291c3aa0eb5c6c8f6b79818aa74/programs/dotnetsdk/install.bash) copies the documented steps to [Register the Microsoft package repository](https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#register-the-microsoft-package-repository).

At the end I test the installation using `dotnet --version`.

[Yesterday's CI run](https://github.com/iainelder/dotfiles/actions/runs/5010823148/jobs/8981047569) worked like this.

```text
+ dotnet --version
7.0.302
```

[Today's](https://github.com/iainelder/dotfiles/actions/runs/5021129495/jobs/9003216146#step:5:366) fails with the error in the title.

```text
+ dotnet --version
The command could not be loaded, possibly because:
  * You intended to execute a .NET application:
      The application '--version' does not exist.
  * You intended to execute a .NET SDK command:
      No .NET SDKs were found.

Download a .NET SDK:
https://aka.ms/dotnet/download

Learn about SDK resolution:
https://aka.ms/dotnet/sdk-not-found
```

Now how do I check the version?

It started happening again on Ubuntu 22.

On 2024-04-02 I had my [last successful CI run](https://github.com/iainelder/dotfiles/actions/runs/8517861317/job/23329069725).

```
+ dotnet --version
7.0.407
```

Since 2024-04-03 [my CI runs fail like this](https://github.com/iainelder/dotfiles/actions/runs/8533683340/job/23376777269#step:5:428):

```
+ dotnet --version
The command could not be loaded, possibly because:
  * You intended to execute a .NET application:
      The application '--version' does not exist.
  * You intended to execute a .NET SDK command:
      No .NET SDKs were found.

Download a .NET SDK:
https://aka.ms/dotnet/download

Learn about SDK resolution:
https://aka.ms/dotnet/sdk-not-found
Error: Process completed with exit code 145.
```

--- a/docs/core/install/linux-package-mixup.md
+++ b/docs/core/install/linux-package-mixup.md
@@ -190,15 +190,29 @@ Configure your package manager to ignore the .NET packages from the distribution
     touch /etc/apt/preferences
     ```
 
-01. Open `/etc/apt/preferences` in an editor and add the following settings, which prevents packages that start with `dotnet`, `aspnetcore`, or `netstandard` from being sourced from the distribution's repository.
+01. Open `/etc/apt/preferences` in an editor and add an apt preferences fragement to prevent packages that start with `dotnet`, `aspnetcore`, or `netstandard` from being sourced from the distribution's repository.
 
-    ```bash
+    ```
     Package: dotnet* aspnet* netstandard*
     Pin: origin "<your-package-source>"
-    Pin-Priority: -10
+    Pin-Priority: -10    
     ```
 
-    Make sure to replace `<your-package-source>` with your distribution's package source, for example, on Ubuntu you may use `archive.ubuntu.com` in the US.
+    Make sure to replace `<your-package-source>` with your distribution's package source.
+
+    Your distribution may have more than one package source.
+
+    For example, on Ubuntu in the US, you may need to use one fragment for `archive.ubuntu.com` and another for `security.ubuntu.com`. Separate each fragment with a blank line like this:
+
+    ```
+    Package: dotnet* aspnet* netstandard*
+    Pin: origin "archive.ubuntu.com"
+    Pin-Priority: -10
+
+    Package: dotnet* aspnet* netstandard*
+    Pin: origin "security.ubuntu.com"
+    Pin-Priority: -10
+    ```
 
     > [!TIP]
     > Use the `apt-cache policy` command to find the source:
@@ -241,6 +255,7 @@ Many other users have reported these problems. The following is a list of those 
   - [Core #4605: cannot run "dotnet new console"](https://github.com/dotnet/core/issues/4605)
   - [Core #4644: Cannot install .NET Core SDK 2.1 on Fedora 32](https://github.com/dotnet/core/issues/4655)
   - [Runtime #49375: After updating to 5.0.200-1 using package manager, it appears that no sdks are installed](https://github.com/dotnet/runtime/issues/49375)
+  - [Installer #16438: The application '--version' does not exist](https://github.com/dotnet/installer/issues/16438)
 
 ## See also
 

